src/input.c

```diff
@@ -990,20 +990,17 @@ static void input_key( int keynum, double value, double kabs, int repeat )
       }
    /* toggle speed mode */
    } else if (KEY("speed") && !repeat) {
+     double newSpeed;
       if ((value==KEY_PRESS) && (!player_isFlag( PLAYER_CINEMATICS_2X ))) {
-         if (player_isFlag(PLAYER_DOUBLESPEED)) {
-            if (!player_isFlag(PLAYER_AUTONAV)) {
-               pause_setSpeed( player_dt_default() );
-               sound_setSpeed( 1. );
-            }
-            player_rmFlag(PLAYER_DOUBLESPEED);
-         } else {
-            if (!player_isFlag(PLAYER_AUTONAV)) {
-               pause_setSpeed( 2. * player_dt_default() );
-               sound_setSpeed( 2. );
-            }
-            player_setFlag(PLAYER_DOUBLESPEED);
-         }
+        printf("Changing player speed %u", player_getSpeed());
+        if(player_getSpeed() < 4){
+          player_setSpeed(player_getSpeed() + 1);
+        } else {
+          player_setSpeed(1);
+        }
+        newSpeed = (double)player_getSpeed();
+        pause_setSpeed( player_dt_default() * newSpeed);
+        sound_setSpeed( newSpeed );
```

Comment:
Indentation is slightly off here (indented by 2 instead of 3); could you fix this, please?

🕵️